src/main/java/org/spongepowered/api/server/Server.java

```diff
@@ -85,7 +88,7 @@
      * @param name The name to get the player from
      * @return The {@link Player} or empty if not found
      */
-    Optional<Player> getPlayer(String name);
+    Optional<? extends Player> getPlayer(String name);
```

Comment:
These always return the server player. raise the lower bound.

Reply:
@liach is correct. RemotePlayer (which is what I'd call MCP's EntityOtherPlayerMP) is purely contained to the ClientWorld. Volume-like methods on Server return Server objects.